odoo/addons/base/security/base_security.xml

```diff
@@ -12,6 +12,16 @@
             <field name="perm_read" eval="False"/>
         </record>
 
+        <record model="ir.rule" id="res_partner_rule">
+            <field name="name">res.partner company</field>
+            <field name="model_id" ref="base.model_res_partner"/>
+            <!-- We exclude partners that have internal users (`partner_share` field) from
+            the multi-company rule because it might interfere with the user's company rule
+            and make some users unselectable in relational fields. This means that partners
+            of internal users are always visible, not matter the company setting. -->
+            <field name="domain_force">['|', '|', ('partner_share', '=', False), ('company_id', 'in', company_ids), ('company_id', '=', False)]</field>
```

Comment:
Shouldn't be the contrary? (it should be visible if the partner is shared?)

```suggestion
            <field name="domain_force">['|', '|', ('partner_share', '=', True), ('company_id', 'in', company_ids), ('company_id', '=', False)]</field>
```

Reply:
No, it's correct as-is.

`partner_share` will be True if the partner has no user or if all its linked users are members of a 'share' group (which can be read as 'all users for that users are portal or public ones').  Partners where `partner_share` is True are definitely the ones we want the multi-company filter applied on.

Apologies for the very confusing "share" terminology...

I just made a quick check (to be sure). There are 3 'user types' groups: internal, portal and public. Both portal and public are marked as shared, so this seems to be ok in all cases.

Reply:
Indeed terminology seems a bit confusing, but I see now. Thanks for clarifying.